### ISSUE TITLE
Weird stuff getting stored in Storage

### ISSUE THREAD

**Issue by double-beep** [Tampermonkey/tampermonkey#1285]
## Script

```js
// ==UserScript==
// @name         Test storage
// @version      0.1
// @author       double-beep
// @match        https://example.com
// @grant        GM_getValue
// @grant        GM_setValue
// @grant        GM_deleteValue
// ==/UserScript==

(function() {
    'use strict';

    GM_setValue('testString', 'testValue');
    GM_setValue('testStringifiedObject', JSON.strinfify({ a: 'b' }));
    GM_setValue('testObject', { a: 'b' });
})();
```

## Expected Behavior

Storage should look like this:

```
{
    "testString": "testValue",
    "testObject": {
        "a": "b"
    }
}
```

## Actual Behavior

Storage looks like this:

```
{
    "ts": 1626420738150,
    "data": {
        "testString": "stestValue",
        "testObject": "o{\"a\":\"b\"}"
    }
}
```

## Specifications

- Chromium: 91.0.4472.164
- **TM Beta 4.14.6141**
- OS: Windows 10.0.19042.1083

Is this intentional?

**Comment by derjanb** [comment]
> Is this intentional?

Yes, it's kind of serialized to a string in order to support all types of storages, even if they have no JSON data but only key string value support.

**Comment by double-beep** [comment]
@derjanb was that introduced in 4.14.6141?

**Comment by derjanb** [comment]
No, it is implemented this way since the beginning. Have you seen something else?

**Comment by double-beep** [comment]
Yes, before that update, the storage tab's textarea showed the "expected behaviour" piece of code. And it still does in TM stable 4.13. This changed in 4.14.6141 for me.

**Comment by derjanb** [comment]
Ah ok. I'll check that.

**Comment by derjanb** [comment]
Should be fixed at 4.14.6142
